app/components/BuyNBTC.tsx

```diff
@@ -142,52 +194,76 @@ export function BuyNBTC() {
 		mode: "all",
 		reValidateMode: "onChange",
 		disabled: isPending || isSuccess || isError,
+		defaultValues: {
+			suiAmount: null,
+		},
 	});
-	const { watch } = buyNBTCForm;
+	const { watch, trigger } = buyNBTCForm;
 	const suiAmount = watch("suiAmount");
 	const amountOfNBTC = Number(suiAmount) / pricePerNBTCInSUI || 0;
 
-	const handleTransaction = useCallback(
-		({ suiAmount }: BuyNBTCForm) => {
-			const suiAmountMist = suiToMist(Number(suiAmount));
-			const transaction = new Transaction();
-			const [coins] = transaction.splitCoins(transaction.gas, [transaction.pure.u64(suiAmountMist)]);
-			transaction.moveCall({
-				target: `${packageId}::${module}::${swapFunction}`,
-				arguments: [
-					transaction.object(vaultId), // Vault object
-					coins, // Coin<SUI> argument
-				],
+	useEffect(() => {
+		if (!account || !suiAmount) return;
+		const suiAmountMist = suiToMist(Number(suiAmount));
+		const txn = new Transaction();
+		txn.setSender(account?.address);
+		const [coins] = txn.splitCoins(txn.gas, [txn.pure.u64(suiAmountMist)]);
+		txn.moveCall({
+			target: `${packageId}::${module}::${swapFunction}`,
+			arguments: [txn.object(vaultId), coins],
+		});
```

Comment:
Create a function to create the TX logic. This helps us update easily in the future.

Reply:
Done